tests/generate_fixtures/generate_target_dnadiff_matrices.py

```diff
@@ -94,41 +106,56 @@ def parse_dnadiff_report(dnadiff_report: Path) -> tuple[int, Decimal, Decimal]:
     return (aligned_bases, query_coverage, avg_identity)
 
 
-# Constructing a matrix where the MD5 hashes of test genomes are used as both column names and index.
-sorted_stems = sorted(file.stem for file in INPUT_DIR.glob("*.f*"))
-aln_lengths_matrix = pd.DataFrame(index=sorted_stems, columns=sorted_stems)
-coverage_matrix = pd.DataFrame(index=sorted_stems, columns=sorted_stems)
-identity_matrix = pd.DataFrame(index=sorted_stems, columns=sorted_stems)
-sim_errors = pd.DataFrame(index=sorted_stems, columns=sorted_stems)
+dnadiff = get_dnadiff()
+print(f"Using nucmer {dnadiff.version} at {dnadiff.exe_path}")
+
+
+# Running comparisons (all vs all)
+inputs = {_.stem: _ for _ in sorted(Path(INPUT_DIR).glob("*.f*"))}
+comparisons = product(inputs, inputs)
+
+# Generate and parse .report files
+for genomes in comparisons:
```

Comment:
If you write this as ``for (query, subject) in genomes:`` you can avoid all the ugly ``genomes[0]`` and ``genomes[1]`` usage.

This will reduce the overall diff as well.